### DISCUSSION TITLE
How to move existing issues into Discussions

### DISCUSSION THREAD

**Discussion by tamlok** [community/community#2952] [Discussions] [answered]
A `discussion` is indeed an `issue`. Could we move an existing issue to be a discussion, since before `Discussions`, I usually used some issues for discussion.

Thanks!

**Comment by yermak** [comment]
> It's very valid request - I have several issues which are questions and general discussions and feedback - it would be just great to keep history of those discussions in relevant place

Update: found this feature. it's already in place

**Reply by tamlok** [reply]
Where to find this feature, please?

**Reply by tamlok** [reply]
Ah, got it!

![image](https://user-images.githubusercontent.com/5763157/101874632-34c6ba80-3bc4-11eb-92e7-4ced85d568f6.png)

This morning when I check it in my phone, I could not find this button since it is at the very bottom behind a long conversation list.

**Reply by bradbell** [reply]
> > It's very valid request - I have several issues which are questions and general discussions and feedback - it would be just great to keep history of those discussions in relevant place
> 
> Update: found this feature. it's already in place

I do not see the choice 'Convert discussion' under 'Transfere issue' for  issue https://github.com/coin-or/CppAD/issues/70 To be specific, what I see is:
![image](https://user-images.githubusercontent.com/37777/103493417-6e1ece80-4dee-11eb-894a-a46db964d82a.png)

**Reply by unixfox** [reply]
> > > It's very valid request - I have several issues which are questions and general discussions and feedback - it would be just great to keep history of those discussions in relevant place
> > 
> > 
> > Update: found this feature. it's already in place
> 
> I do not see the choice 'Convert discussion' under 'Transfere issue' for issue [coin-or/CppAD#70](https://github.com/coin-or/CppAD/issues/70) To be specific, what I see is:
> ![image](https://user-images.githubusercontent.com/37777/103493417-6e1ece80-4dee-11eb-894a-a46db964d82a.png)

I think it got removed a couple of days ago for an unknown reason. At least I had the option one week ago.

**Reply by bradbell** [reply]
When you did this, did it remove the issue, or did in keep it with a link to the discussion where it had been moved. I think removing the issue would be a bad idea because some people might look for it where it used to be.

**Reply by shadowtime2000** [reply]
Running into the same problem, I just see this for [eta-dev/eta#76](https://github.com/eta-dev/eta/issues/76).
![image](https://user-images.githubusercontent.com/66655515/103611317-e1037480-4ed6-11eb-8421-ab7734a260e8.png)

**Reply by unixfox** [reply]
> When you did this, did it remove the issue, or did in keep it with a link to the discussion where it had been moved. I think removing the issue would be a bad idea because some people might look for it where it used to be.

It removed the issue and then converted the issue in the same way as it was before but in a discussion so the comments are still there with their author.

**Reply by beauby** [reply]
Same here, it was available a week or so ago, but not anymore. Hopefully it will be back soon?

**Reply by michaelklishin** [reply]
Looks like this feature was removed. This has largely eliminated the benefit of having discussions for my team: we routinely see users file issues for what are questions or requests for troubleshooting help, no issues with any amount of evidence. We would typically convert those to discussions but no longer have this option 😭

**Reply by jef** [reply]
Probably will get reimplemented at some point. Discussions is still in beta and is subject to change. I'm sure it won't be gone too long, it's a very helpful tool.

**Reply by ericblade** [reply]
yeah i did this with an issue when i first got the feature, just went to start weeding discussion issues over into discussions now, and the entire convert issue to discussion feature is gone :(

**Reply by ericblade** [reply]
The bulk convert by label is also missing

**Reply by m7kvqbe1** [reply]
Please bring this back!!

**Comment by rwightman** [comment]
I used to see this option, and have used it a few times but now it's not there.. what happened?

**Reply by shadowtime2000** [reply]
It looks like it was removed, I have no idea why...

**Reply by jef** [reply]
https://github.com/github/feedback/discussions/2952#discussioncomment-263319

**Reply by ericblade** [reply]
there's a post somewhere here that says it was temporarily removed to address bugs

**Comment by evi-liu** [comment]
Hi all, we've temporarily disabled conversion from Issues to Discussion due to a high priority bug that was found. Our team is hard at work at resolving this bug to enable the capability again.

**Reply by ssbarnea** [reply]
@evi-liu Thanks for letting us know. I am looking forward to see the feature back.

**Reply by yecol** [reply]
Looking forward the return of this feature. :)

**Reply by roymx** [reply]
@evi-liu +1 -- looking forward to use it again!

**Reply by evi-liu** [reply]
Thanks for your patience, y'all!

**Reply by anjola-adeuyi** [reply]
Almost a year, looking foward. 🚀

**Reply by simsong** [reply]
still waiting.

**Comment by evi-liu** [comment, marked as answer]
Hi all! We now have the fix ready for conversions. If you'd like to be flagged in early so you can start converting again, do you mind writing to this thread with the repository to be added? 
* The issue will be closed and locked.
* Title, description, and author will be the same as the issue.
* All comments and reactions should be the same as the issue.

Please let me know if you encounter any unexpected behavior. We will continue to monitor closely. And a huge THANK YOU for your patience through this bug!

**Reply by rwightman** [reply]
These two please
* https://github.com/rwightman/pytorch-image-models
* https://github.com/rwightman/efficientdet-pytorch

**Reply by evi-liu** [reply]
@rwightman both added!

**Reply by CaselIT** [reply]
Could also please add https://github.com/sqlalchemy/sqlalchemy
Thanks!

**Reply by evi-liu** [reply]
@CaselIT added!

**Reply by fda77** [reply]
Please add https://github.com/Freetz-NG/freetz-ng

**Reply by evi-liu** [reply]
@fda77 added!

**Reply by unixfox** [reply]
Could you add https://github.com/searx/searx?
Thank you.

**Reply by evi-liu** [reply]
@unixfox added!

**Reply by roymx** [reply]
Please add https://github.com/InterDigitalInc/AdvantEDGE
Thanks @evi-liu !

**Reply by bradbell** [reply]
Please add https://github.com/coin-or/CppAD

**Reply by fda77** [reply]
Feedback:
 - After converting Issue to Discussion , the Issue keeps there, so i just deleted it
 - It would be nice to convert a discusstion to an issue, users are not always sure which is better

**Reply by evi-liu** [reply]
@roymx and @bradbell your repos have been added!

**Reply by maxandersen** [reply]
jbangdev/jbang

**Reply by evi-liu** [reply]
@maxandersen added!

**Reply by jef** [reply]
https://github.com/jef/streetmerchant

Thank you!

**Reply by evi-liu** [reply]
@jef added!

**Reply by jcheron** [reply]
Thanks for https://github.com/phpMv/ubiquity

**Reply by shurup** [reply]
Please add these repos as well:
* https://github.com/flant/grafana-statusmap
* https://github.com/flant/addon-operator
* https://github.com/werf/werf

TIA!

**Reply by billz** [reply]
https://github.com/billz/raspap-webgui/
Many thanks!

**Reply by ssbarnea** [reply]
https://github.com/ansible-community/ansible-lint
https://github.com/ansible-community/molecule

**Reply by beutlich** [reply]
https://github.com/modelica/ModelicaStandardLibrary

**Reply by emoacht** [reply]
Could you please add https://github.com/emoacht/SnowyImageCopy
Thank you.

**Reply by seanlaw** [reply]
Please add https://github.com/TDAmeritrade/stumpy
Thank you!

**Reply by evi-liu** [reply]
all the repositories requested above this comment have been added!

**Reply by marchbold** [reply]
Could you add ours too? https://github.com/Gamua/Adobe-Runtime-Support
Thanks @evi-liu

**Reply by mariusoe** [reply]
It would be greate if you could add our repo, too: https://github.com/NovatecConsulting/novatec-service-dependency-graph-panel

**Reply by edgarriba** [reply]
this one too: https://github.com/kornia/kornia

**Reply by evi-liu** [reply]
@marchbold @mariusoe @edgarriba added!

**Reply by emteere** [reply]
Please add https://github.com/NationalSecurityAgency/ghidra
Thanks @evi-liu

**Reply by evi-liu** [reply]
@emteere done!

**Reply by simeydotme** [reply]
Please add https://github.com/simeydotme/svelte-range-slider-pips/ for converting issue to discussion
Thanks @evi-liu  :)

**Reply by JorjMcKie** [reply]
please also add https://github.com/pymupdf/PyMuPDF
Thank you!

**Reply by bkolobara** [reply]
Could you please add: https://github.com/lunatic-solutions/lunatic. Thanks a lot @evi-liu!

**Reply by msberends** [reply]
https://github.com/msberends/AMR please! Thanks!

**Reply by applejag** [reply]
https://github.com/jilleJr/Newtonsoft.Json-for-Unity

**Reply by avital** [reply]
Hi! We'd like "convert to discussions" re-enabled for https://github.com/google/flax -- we use GitHub discussions heavily and a lot of users still open issues accidentally, it's quite rude to ask them to close the issue and re-open as a discussion. Thanks!

**Reply by jrbarnes9** [reply]
Please add: https://github.com/ONSdigital/design-system/

**Reply by bpabiszczak** [reply]
Please add https://github.com/YetiForceCompany/YetiForceCRM

**Reply by adriankumpf** [reply]
Could you please add: https://github.com/adriankumpf/teslamate Thank you!

**Reply by mifi** [reply]
https://github.com/mifi/lossless-cut thanks

**Reply by bemasher** [reply]
I'd like to give this a go as well: https://github.com/bemasher/rtlamr

**Reply by opencrx** [reply]
Could you please add https://github.com/opencrx/opencrx and https://github.com/openmdx/openmdx

**Reply by NiklasMerz** [reply]
Could you please add: https://github.com/NiklasMerz/cordova-plugin-fingerprint-aio

Really love this feature.

**Reply by kryskool** [reply]
can you enable this feature on https://github.com/modoboa/modoboa 
we use it until it removed

Regards,

**Reply by evi-liu** [reply]
Please see https://github.com/github/feedback/discussions/2952#discussioncomment-293802. Since we're starting unroll the fix now, I'm going to stop manually flagging in repositories. You should see the conversion ability in your repositories by end of day.

**Reply by m7kvqbe1** [reply]
https://github.com/Royal-Navy/design-system

**Reply by evi-liu** [reply]
Hi all, unfortunately we were not able to complete rolling out the conversion ability due to some unexpected bugs popping up again. In the meantime, I've flagged in all of the above repositories so you can now convert issues to discussions again. Please let me know if any unexpected behavior occurs.

**Reply by PureWeen** [reply]
@evi-liu can you flag the following repository please? https://github.com/dotnet/maui

Excited this feature is coming back! Discussions have been great for helping us keep Issues betters organized and focused.

**Reply by reesericci** [reply]
@evi-liu  Please flag coredns. :) [coredns/coredns](github.com/coredns/coredns)

**Reply by sighingnow** [reply]
please add :) https://github.com/alibaba/GraphScope @evi-liu

**Reply by rabitt** [reply]
Please add https://github.com/mir-dataset-loaders/mirdata @evi-liu . Thank you!

**Reply by ewels** [reply]
Please add https://github.com/ewels/MultiQC @evi-liu  - thanks!

**Reply by rdeutz** [reply]
please add https://github.com/joomla/joomla-cms also -- Thanks!

**Reply by kjmartens** [reply]
Please add https://github.com/solspace/craft-express-forms 🙂

**Reply by evi-liu** [reply]
✅ all repos in this thread have been added! you may now convert issues again.

**Reply by christianfelicite** [reply]
this one please : https://github.com/TheFeloDevTeam/FeloFamilySite

**Reply by cortinico** [reply]
Hey 👋 Please add [detekt/detekt](https://github.com/detekt/detekt)

**Reply by WouterJD** [reply]
please add https://github.com/WouterJD/FortiusANT

**Reply by saul** [reply]
Please add https://github.com/saul/demofile

**Reply by retorquere** [reply]
https://github.com/retorquere/zotero-sync and https://github.com/retorquere/zotero-better-bibtex/ please!

**Reply by erwincoumans** [reply]
Yes, https://github.com/bulletphysics/bullet3

**Reply by sTiKyt** [reply]
https://github.com/Open-Security-Group-OSG/HiddenEyeReborn

@evi-liu this one please

**Reply by ErikSin** [reply]
This one: https://github.com/digidem/mapeo-mobile and this one: https://github.com/digidem/mapeo-desktop please

**Reply by liamsi** [reply]
https://github.com/celestiaorg/celestia-node/issues/323

**Reply by nbouvrette** [reply]
https://github.com/Avansai/next-multilingual thanks

**Reply by Taitava** [reply]
I cannot unlock an issue that has been converted into a discussion: https://github.com/Taitava/obsidian-shellcommands/issues/88
(see my comment below: https://github.com/github/feedback/discussions/2952#discussioncomment-1710762)

**Reply by xeno6696** [reply]
https://github.com/ESAPI/esapi-java-legacy

**Reply by stchris** [reply]
I'd love to use this on https://github.com/stchris/untangle

**Reply by adamlui** [reply]
I'd like to move this feature request from issues to discussion https://github.com/adamlui/userscripts/issues/2

**Reply by adamlui** [reply]
Never mind, I was able to convert using this button in teh sidebar!
![image](https://user-images.githubusercontent.com/10906554/213171840-7b952b85-2b74-4ed5-83c7-b2c2c4925885.png)

**Reply by swirle13** [reply]
@evi-liu Your documentation states that future issues with the selected label(s) will automatically be converted into a discussion, yet the UI when selecting this option specifically states that it will _not_ automatically convert issues with label "X" to a discussion? Why is this the case? How can we automate converting any issue with discussion labels into a discussion? More info in this post https://github.com/orgs/community/discussions/52785

**Comment by NiklasMerz** [comment]
I found this documentation about converting issues based on labels but cannot find it right now: https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions-in-your-repository#converting-issues-based-on-labels

Will this be rolled out in the future? Looks really helpful.

**Reply by evi-liu** [reply]
Conversions are temporarily disabled since we found a high priority bug (https://github.com/github/feedback/discussions/2952#discussioncomment-267762).

**Comment by evi-liu** [comment]
Hi all, we've now rolled out conversion capability again to 50% of repositories. We're hoping to enable to 100% of repositories by the end of today, so stay tuned and you should be able to convert again very, very soon. Once again, so sorry for the inconvenience this has caused and let me know if you find any unexpected behavior.

**Reply by Harmon758** [reply]
Once an issue is converted to a discussion, its labels can't be modified. Adding or removing a label to/from the issue won't error and will appear as if it was successful in the "Apply labels to this issue" dropdown, but upon refreshing the page, the labels remain unchanged. Is this intentional?

Also, after an issue is converted to a discussion, unlocking it doesn't seem to do anything. It appears as if its still locked in that there's no Write/Preview editor to add a comment. Is this intentional?

**Reply by evi-liu** [reply]
Thank you for reporting. We will look into this. Out of curiosity, what would be the intention behind unlocking an issue that has already been converted to a discussion?

**Reply by Harmon758** [reply]
Well, in this case, I was just testing the first issue and seeing if unlocking the issue allowed modifying the labels, which it does not, but I suppose it would be useful in cases where the discussion results in a need to revert back to an issue or where the intention behind the conversion is to create a branching discussion about the issue while the issue itself remains separate. Although, at the moment, I don't think there's a way to reopen an issue either, after it's been converted to a discussion.

**Reply by wfro64** [reply]
Thank you very much. It looks like the conversion capability has not been rolled out to all projects. https://github.com/opencrx/opencrx does not have this feature, but https://github.com/opencrx/opencrx-documentation does.

**Reply by ndbeals** [reply]
Hello @evi-liu, thank you for such fast and open communication.

Would you mind enabling conversions from issues to discussions on this repository? https://github.com/ndbeals/keep-exporter 

Thanks

**Reply by m7kvqbe1** [reply]
Please could you also enable it for this repo: https://github.com/Royal-Navy/design-system

Many thanks!

**Reply by evi-liu** [reply]
the repos above have been flagged in and have the capability to convert again!

**Reply by Harmon758** [reply]
Converted issues seem to be able to be labeled afterwards now.
I've also noticed that there doesn't seem to be a way to delete a converted issue, even though [that was part of the initial proposal for changing it to being locked instead of deleted](https://github.com/github/feedback/discussions/3105#discussioncomment-274982).

**Reply by evi-liu** [reply]
@Harmon758 Thank you for the report! I've passed it along to the team.

**Reply by sighingnow** [reply]
Hi @evi-liu,could please enable this repo https://github.com/alibaba/GraphScope?

Thanks a lot!

**Reply by shadowtime2000** [reply]
@evi-liu Could you please enable `alephjs/aleph.js` and `eta-dev/eta`?

**Reply by JC5** [reply]
Hi! I am still looking for the button but it doesn't seem to be present at the moment. I created Discussions only a few days ago. Is there something I can do?

https://github.com/firefly-iii/firefly-iii/issues/4306

**Reply by evi-liu** [reply]
✅  the repos above have been flagged in and now have conversion capability

**Reply by NathanNovak** [reply]
@evi-liu can you add https://github.com/StraboSpot/StraboSpot-Mobile to the have conversion capability? Then and short description of how to convert.

**Reply by reesericci** [reply]
@evi-liu can you please flag coredns/coredns

**Reply by dopplershift** [reply]
@evi-liu I would appreciate flagging in https://github.com/Unidata/MetPy

**Reply by evi-liu** [reply]
✅  repos above have been flagged in

**Reply by wangchucheng** [reply]
@evi-liu Please flag https://github.com/wangchucheng/hugo-eureka if possible, thanks!

**Reply by RichardWallis** [reply]
Is the conversion capability still missing because of problems, or it has not yet had 100% roll out?

If it is the latter, can you flag https://github.com/schemaorg/schemaorg

**Reply by Taitava** [reply]
> Thank you for reporting. We will look into this. Out of curiosity, what would be the intention behind unlocking an issue that has already been converted to a discussion?

@evi-liu It seems that it's still not possible to unlock issues after they have been converted to discussions.

Example issue in my repo: https://github.com/Taitava/obsidian-shellcommands/issues/88

I have converted this to a discussion, because it's a feature request and I wanted to first have a conversation in a separate place, and later use the issue again so that it will contain a summary of the discussion, i.e. decisions on how the feature will be implemented. But it's not possible. I have clicked _Unlock conversation_, but there's still no way to add new comments or edit the title (I'd like to change the title to better indicate how the feature will actually be implemented).

**Reply by Taitava** [reply]
@evi-liu ? 🙂

**Reply by Taitava** [reply]
@github-staff @evi-liu I'd still like to get this solved. 🙂

**Reply by Taitava** [reply]
Still waiting... 🙂

**Comment by davestephens** [comment]
There is some chatter above from @evi-liu  about this being rolled out to 100% of repos nearly three weeks ago. Was that done, or do I need to request it gets added to my repo too? An official update from GitHub would be appreciated, thanks!

**Reply by Harmon758** [reply]
See https://github.com/github/feedback/discussions/2952#discussioncomment-300956.

**Reply by evi-liu** [reply]
beat me to it:) https://github.com/github/feedback/discussions/2952#discussioncomment-328878

**Comment by bradbell** [comment]
I moved an issue to a discussion. But the issue is closely related to another discussion. The two discussions have different titles because they are different from the user's perspective. It is as if I would like two threads under one over arching discussion. Is there some way to do  this ?

**Comment by evi-liu** [comment]
**Update:** We've now rolled out `convert issue to discussion` to 100% of repositories in the public beta. 

Thank you all for your patience with this bug! It's been a gnarly one for our team. We've added more robust logging as well as verification checks to retry the conversion process if it fails. Hopefully, this mitigates any risk with conversions. We're aware that the current locking mechanism for converted issues is sub-optimal, so once we feel more confident that the underlying bug has been fixed, our intention is to revert back to deleting converted issues.

**Reply by RichardWallis** [reply]
Can we have https://github.com/schemaorg/suggestions-questions-brainstorming added please.

**Reply by JirkaDellOro** [reply]
What do I need to do to use the public beta?

**Reply by retorquere** [reply]
It doesn't actually move the discussion though. It creates an issue based on the discussion, and you lose authorship of comments, and the participants aren't automatically watching the issue.

**Comment by 520299XX** [comment]
Horizon 5 is updated, when is the Brush Full Car app updated.

**Comment by bradbell** [comment]
Looking at Converting and Issue to a Discussion on
https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion

It says 'In the right margin of an issue, click Convert to discussion.' but I do not see that for one of my repositories.
I do see 'Delete Issue' and some other options.

Is there something I need to do to make this available for a specific repository that I manage ?

**Reply by poszu** [reply]
I had this problem too (I didn't have "convert to discussion" button). I had to go to the _Discussions_ tab, click "Get started" and create the first post (GH has a prefilled "welcome" template). After posting it the button to convert issue to a discussion appeared on all my issues.

**Reply by bradbell** [reply]
@poszu Thanks. Your suggestion worked for me. Actually, I had to start with the settings and select Discussions and then following the instructions.

**Comment by V2U-Services** [comment]
Existing issues cannot be converted directly to GitHub Discussions. Instead, create a new discussion by hand and include the pertinent issues in it. Copy important details and add context. GitHub Actions or scripts can be used as an alternative to automate this, but manual linking guarantees greater organization and supervision.

**Reply by poszu** [reply]
They can be converted with a button on the right side of the issue:
![image](https://github.com/user-attachments/assets/854b2b6f-065c-4875-9b7a-342fe3328e7b)

**Reply by Taitava** [reply]
However, that "conversion" creates a new discussion and locks the original issue. (At least that's how it worked the last time I used it).

For me, I'd prefer an option to really _convert_ an issue to a discussion - that is, to use the same issue # number, just change the type of the issue to a discussion. No need to create anything new. 🙂

**Comment by 520299XX** [comment]
不太明白什么意思



发自我的iPhone


------------------ Original ------------------
From: Bartosz Różański ***@***.***&gt;
Date: Thu,Jan 2,2025 5:51 PM
To: community/community ***@***.***&gt;
Cc: 520299XX ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [community/community] How to move existing issues intoDiscussions (Discussion #2952)





 
They can be converted with a button on the right side of the issue:
 image.png (view on web)

**Comment by 520299XX** [comment]
我想知道，能不能直接修改存档文件，直接添加全车，cp，旋转



发自我的iPhone


------------------ Original ------------------
From: Jarkko Linnanvirta ***@***.***&gt;
Date: Sat,Jan 11,2025 5:01 PM
To: community/community ***@***.***&gt;
Cc: 520299XX ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [community/community] How to move existing issues intoDiscussions (Discussion #2952)





 
However, that "conversion" creates a new discussion and locks the original issue. (At least that's how it worked the last time I used it).
 
For me, I'd prefer an option to really convert an issue to a discussion - that is, to use the same issue # number, just change the type of the issue to a discussion. No need to create anything new. 🙂

**Comment by axu2** [comment]
I created a discussion from an issue, which locked the original issue. I actually prefer the original issue format, is there any way to unlock the original issue? I get an error when I try to unlock.

- https://github.com/orgs/community/discussions/2861

EDIT: delete the converted discussion lets you unlock the issue.
